tests/test_cli.py

```diff
@@ -0,0 +1,410 @@
+        self.assertEqual(r.out.strip(), "File created.")
+        self.assertIn(SETTINGS_FN, r.files)
+        data = r.files[SETTINGS_FN].get_written_yaml()
+        self.assertIn('docker', data)
+        self.assertIn('timeout', data['docker'])
+        self.assertEqual(data['docker']['timeout'], 100)
+
+    def test_set_wrong_type(self):
+        r = self.command_test("config -g set docker.tls_verify=2", exit_code=1)
+        self.assertEqual(
+            "docker.tls_verify should be of type 'boolean', but was 'str'.", r.err.strip())
+        r = self.command_test("config -g set docker.timeout=a", exit_code=1)
+        self.assertEqual(
+            "docker.timeout should be of type 'integer', but was 'str'.", r.err.strip())
+
+    def test_set_not_selected(self):
+        r = self.command_test("config -g -p set a=b", exit_code=1)
+        self.assertEqual("Choose either global settings or project settings", r.err.strip())
+
+    def test_set_wrong_format(self):
+        r = self.command_test("config -g set docker.tls_verify: True", exit_code=1)
+        self.assertEqual("Give values in format 'key=val'.", r.err.strip())
+
+
+class TestConfigRemoveAction(CliTestCase):
+    SETTINGS = {
+        'version': '1.0',
+        'docker': {'tls_verify': True}
+    }
+
+    def test_normal_rm(self):
+        r = self.command_test("config -g remove docker.tls_verify", settings=self.SETTINGS)
+        self.assertEqual("File successfully edited.", r.out.strip())
+
+    def test_rm_empty_file(self):
+        r = self.command_test("config -g remove docker.test")
+        self.assertEqual(
+            "Cannot delete from config because config file doesn't exist.", r.out.strip())
+
+    def test_rm_nonexistant(self):
+        r = self.command_test("config -g remove docker.test", settings=self.SETTINGS)
+        self.assertIn("Key docker.test doesn't exist in config.", r.out)
+        self.assertIn("No changes in the file.", r.out)
+
+    def test_rm_config_not_selected(self):
+        r = self.command_test("config -g -p remove a=b", exit_code=1)
+        self.assertEqual("Choose either global settings or project settings", r.err.strip())
+
+
+class TestStepListAction(CliTestCase):
+
+    def test_step_list(self):
+        config = dict(HELLO_CONFIG)
+        config['steps'] = config['steps'] + [{'img': 'hei-maailma', 'name': 'moi'}]
+        r = self.command_test("step list", config=config)
+        self.assertIn("ID  NAME  IMAGE", r.out)
+        self.assertIn(" 0. hello hello-world", r.out)
+        self.assertIn(" 1. moi   hei-maailma", r.out)
+
+    def test_step_list_empty(self):
+        r = self.command_test("step list", config={'version': '2.0'})
+        self.assertEqual("The project config has no steps.", r.out.strip())
+
+
+class TestStepAddAction(CliTestCase):
+
+    def test_step_add(self):
+        r = self.command_test(
+            "step add hello-world --name new_step "
+            "--cmd \"make touchrst html\" "
```

Comment:
You should use different quotes on outer and inner strings. e.g. `"-- cmd 'make touchrst html' "`.